### ISSUE TITLE
Group Assignment Bug 

### ISSUE THREAD

**Issue by small-fork** [johngodley/redirection#3364]
This screencast will explain far better than my written explanation: https://www.loom.com/share/bb4edab0f5ff434082e8c8ce7ed9f847

### Summary

I have had the group links are assigned to "disappear" from the admin columns when navigating through a selected group. When I then go to edit any individual link, it shows to be in the wrong group, even though if I go to the group list and select "view redirects" it does filter to the links in that group, but doesn't show the correct group in the column or if editing the link.

Additionally, the proper group is not selectable from the drop-down in order to try to "re-assign" the redirect. 

### Actual Behavior

I navigate to Groups, then click "view redirects". I'm paging through that group and I have the group selected in the table "custom display". At some point the group stops showing. I then notice that if I edit any individual redirect, it's showing the wrong group, though still correct if filtered for the original group (again please see the screencast as this is hard to describe)

### Environment

Wordpress: 5.9.3
Browser: Vivaldi
Redirection: 5.2.3

**Comment by small-fork** [comment]
UPDATE: I have determined the exact trigger. It doesn't make sense, but this bug occurs if you change the "Screen Options" count for the list.

Please see: https://www.loom.com/share/b4ab7299c6514d6bb7612141029cec47

**Comment by small-fork** [comment]
UPDATE: Apologies for yet another update, but I have narrowed down the behavior. It appears that when you reduce the "Screen Options" number to something BELOW/Less than the number of Groups you have, any Groups that are on the 2nd page on the Groups tab will have this behavior of no longer being visible within the Redirects table itself. 

See: https://www.loom.com/share/9a54919784864e939609c9ebb8443c4c

**Comment by johngodley** [comment]
Yep, this is a known problem. For now you will need to have it set to the same number as your groups.

**Comment by johngodley** [comment]
This will be fixed in the next release.
